enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,687 @@
+not be routes propagated into the BGP network for unselected nodes. Users should expect disruption when the transport of
+a network is changed.
+
+#### BGP Configuration
+
+When OVN-Kubernetes detects that a FRRConfiguration has been created that has a corresponding and valid FRRNodeState,
+OVN-Kubernetes will then use RouteAdvertisements CR create a corresponding FRRConfiguration. The following examples will
+use an environment where a user has created an FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  creationTimestamp: "2024-06-11T14:22:37Z"
+  generation: 1
+  name: metallb-ovn-worker
+  namespace: metallb-system
+  resourceVersion: "1323"
+  uid: 99b64be3-4f36-4e0b-8704-75aa5182d89f
+  labels:
+    routeAdvertisements: default
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+  nodeSelector:
+    matchLabels:
+       kubernetes.io/hostname: ovn-worker
+```
+
+OVNKube-Controller will check that if this FRRConfiguration applies to its node, ovn-worker. For this example, a user-defined
+network named "blue", has been created with a network of 10.0.0.0/16, and a matching vrf exists in the Linux host. The slice
+of this supernet that has been allocated to node ovn-worker is 10.0.1./0/24. 
+
+##### Example 1: Advertising pod IPs from a user-defined network over BGP
+
+![](images/bgp_novpn_advertisements.png)
+
+In this example a user wants to expose network blue outside their OCP cluster so that pod IPs are reachable on the
+external network. The admin has creates the following RouteAdvertisements CR for the blue tenant:
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+  advertisements:
+    podNetwork: true
+  networkSelector:
+    matchLabels: 
+      k8s.ovn.org/metadata.name: blue
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/hostname: ovn-worker
+  frrConfigurationSelector:
+    matchLabels:
+      routeAdvertisements: default
+```
+
+OVNKube-Controller will now see it needs to generate corresponding FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: route-advertisements-blue
+  namespace: metallb-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      vrf: blue
+      prefixes:
+        - 10.0.1.0/24
+    - asn: 64512
+      neighbors:
+         - address: 172.18.0.5
+           asn: 64512
+           toAdvertise:
+              allowed:
+                 prefixes:
+                    - 10.0.1.0/24
+  raw:
+    rawConfig: |-
```

Comment:
Let's add support for route leaking and get rid of raw config

Reply:
I'll mention the rawConfig above should be implemented as an API in FRR-K8S. Is that good enough?

Reply:
sure!